common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/azureactivedirectory/AzureActiveDirectoryAuthorizationRequest.java

```diff
@@ -22,34 +22,258 @@
+                mAuthorizationEndpoint,
+                createAuthorizationRequestParameters());
+        Logger.infoPII(TAG, mCorrelationId.toString(), "Request uri to authorize endpoint is: " + authorizationUrl);
+        return authorizationUrl;
+
+    }
+
+    private Map<String, String> createAuthorizationRequestParameters() throws UnsupportedEncodingException, ClientException {
+        final Map<String, String> requestParameters = new HashMap<>();
+        requestParameters.put(AuthenticationConstants.OAuth2.RESPONSE_TYPE, AuthenticationConstants.OAuth2.CODE);
+        requestParameters.put(AuthenticationConstants.OAuth2.CLIENT_ID,
+                URLEncoder.encode(getClientId(), ENCODING_UTF8));
+        requestParameters.put(RESOURCE, URLEncoder.encode(mResource, ENCODING_UTF8));
+        requestParameters.put(AuthenticationConstants.OAuth2.REDIRECT_URI,
+                URLEncoder.encode(getRedirectUri(), ENCODING_UTF8));
+        requestParameters.put(AuthenticationConstants.OAuth2.STATE, encodeProtocolState());
+        requestParameters.put(AuthenticationConstants.OAuth2.SCOPE, SCOPE_OPENID_VALUE);
+
+        // append device and platform info in the query parameters
+        requestParameters.put(ADAL_ID_PLATFORM, ADAL_ID_PLATFORM_VALUE);
+        requestParameters.put(ADAL_ID_VERSION, URLEncoder.encode(mLibraryVersion, ENCODING_UTF8));
+        requestParameters.put(ADAL_ID_OS_VER, URLEncoder.encode(String.valueOf(Build.VERSION.SDK_INT), ENCODING_UTF8));
+        requestParameters.put(ADAL_ID_DM, URLEncoder.encode(android.os.Build.MODEL, ENCODING_UTF8) );
+
+        if (!StringExtensions.isNullOrBlank(mLoginHint)) {
+            requestParameters.put(LOGIN_HINT, URLEncoder.encode(mLoginHint, ENCODING_UTF8) );
+        }
+
+        if (mCorrelationId != null) {
+            requestParameters.put(CLIENT_REQUEST_ID, URLEncoder.encode(mCorrelationId.toString(), ENCODING_UTF8));
+        }
+
+        // Setting prompt behavior to always will skip the cookies for webview.
+        // It is added to authorization url.
+        if (mPromptBehavior == AADPromptBehavior.Always) {
+            requestParameters.put(QUERY_PROMPT, URLEncoder.encode(QUERY_PROMPT_VALUE, ENCODING_UTF8));
+        } else if (mPromptBehavior == AADPromptBehavior.REFRESH_SESSION) {
+            requestParameters.put(QUERY_PROMPT, URLEncoder.encode(QUERY_PROMPT_REFRESH_SESSION_VALUE, ENCODING_UTF8));
+        }
+
+        if (null != mPKCEChallenge) {
+            requestParameters.put(CODE_CHALLENGE, URLEncoder.encode(mPKCEChallenge.getCodeChallenge(), ENCODING_UTF8));
+            // The method used to encode the code_verifier for the code_challenge parameter.
+            requestParameters.put(CODE_CHALLENGE_METHOD, URLEncoder.encode(mPKCEChallenge.getCodeChallengeMethod(), ENCODING_UTF8));
+        }
+
+        // Reading extra qp supplied by developer. append haschrome=1 if developer does not pass as extra qp
```

Comment:
Can we javadoc _why_ we're appending `haschrome=1` and preferably what it means?

By appending `haschrome` are we asserting to AAD that the developer doesn indeed have Chrome installed? Do we need to actually check this? If we do, where does that happen?